File: app/src/main/java/com/ephirium/storyline/feature/ui/CreateFragment.java
package com.ephirium.storyline.feature.ui;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.appcompat.view.menu.MenuBuilder;
import androidx.fragment.app.Fragment;

import android.util.Log;
import android.view.ActionMode;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import com.ephirium.storyline.R;
import com.ephirium.storyline.databinding.FragmentCreateBinding;

public class CreateFragment extends Fragment {

    private FragmentCreateBinding binding;

    public CreateFragment() {
    }

    public static CreateFragment newInstance() {
        return new CreateFragment();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentCreateBinding.inflate(getLayoutInflater());

        binding.storyText.setCustomSelectionActionModeCallback(new ActionMode.Callback2() {
            @Override
            public boolean onCreateActionMode(ActionMode mode, Menu menu) {
                MenuInflater inflater = mode.getMenuInflater();
                inflater.inflate(R.menu.text_select_menu, menu.addSubMenu("Format"));
                return true;
            }

            @Override
            public boolean onPrepareActionMode(ActionMode mode, Menu menu) {

                return true;
            }

            @Override
            public boolean onActionItemClicked(ActionMode mode, MenuItem item) {

                return true;
            }

            @Override
            public void onDestroyActionMode(ActionMode mode) {

            }
        });

        return binding.getRoot();
    }
}

File: app/src/main/java/com/ephirium/storyline/ui/fragment/CreateFragment.kt
package com.ephirium.storyline.ui.fragment;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.ActionMode;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import by.kirich1409.viewbindingdelegate.viewBinding

import com.ephirium.storyline.R;
import com.ephirium.storyline.databinding.FragmentCreateBinding;

class CreateFragment : Fragment(R.layout.fragment_create) {

    private val binding : FragmentCreateBinding by viewBinding()

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.storyText.customSelectionActionModeCallback = object : ActionMode.Callback2(){
            override fun onCreateActionMode(mode: ActionMode?, menu: Menu?): Boolean {
                val inflater = mode?.menuInflater;
                inflater?.inflate(R.menu.text_select_menu, menu?.addSubMenu("Format"));
                return true;
            }

            override fun onPrepareActionMode(mode: ActionMode?, menu: Menu?) = true

            override fun onActionItemClicked(mode: ActionMode?, item: MenuItem?): Boolean {

                return true
            }

            override fun onDestroyActionMode(mode: ActionMode?) { }

        }
    }
}